WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/FeeOrDiscountLineDetailsViewModel.swift

```diff
@@ -2,11 +2,82 @@ import SwiftUI
 import Yosemite
 import WooFoundation
 
-class FeeLineDetailsViewModel: ObservableObject {
+protocol FeeOrDiscountLineDetailsStringsProvider {
+    var navigationTitle: String { get }
+    var removeButtonTitle: String { get }
+    var doneButtonAccessibilityIdentifier: String { get }
+    var fixedAmountFieldAccessibilityIdentifier: String { get }
+}
+
+private struct StringsProviderFactory {
+    static func stringsProvider(from type: FeeOrDiscountLineDetailsViewModel.LineType, isExistingLine: Bool) -> FeeOrDiscountLineDetailsStringsProvider {
+        switch type {
+        case .discount:
+            return DiscountStringsProvider(isExistingLine: isExistingLine)
+        case .fee:
+            return FeeStringsProvider(isExistingLine: isExistingLine)
+        }
+    }
+}
+
+private struct DiscountStringsProvider: FeeOrDiscountLineDetailsStringsProvider {
+    let isExistingLine: Bool
+
+    var navigationTitle: String {
+        isExistingLine ? Localization.discount : Localization.addDiscount
+    }
+
+    var removeButtonTitle: String {
+        Localization.remove
+    }
+
+    var doneButtonAccessibilityIdentifier: String {
+        "add-discount-done-button"
+    }
+
+    var fixedAmountFieldAccessibilityIdentifier: String {
+        "add-discount-fixed-amount-field"
+    }
+
+    enum Localization {
+        static let addDiscount = NSLocalizedString("Add Discount", comment: "Title for the Discount screen during order creation")
+        static let discount = NSLocalizedString("Discount", comment: "Title for the Discount Details screen during order creation")
+        static let remove = NSLocalizedString("Remove Discount", comment: "Title for the Remove button in Details screen during order creation")
+    }
+}
```

Comment:
Should we move those to a private extension, and the static `NSLocalizedStrings` to the existing `Localization` enum within `FeeOrDiscountLineDetailsViewModel`?